### PR TITLE
Fix scatter plots to render 0-valued metrics

### LINKED ISSUE #13659: [BUG] Runs with 0-valued metrics don't appear in experiment list scatter plots
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

2.17.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS Sonoma 14.5
- **Python version**: 3.11
- **yarn version, if running the dev UI**: N/A

### Describe the problem

Runs with 0-valued metrics don't appear in experiment list scatter plots. 0-valued params work as expected. There are 5 runs logged with (0,4), (1, 3), (2, 2), (3, 1), and (4, 0) as the param, metric pairs. Note that only 4 dots appear
Screenshot 2024-11-04 at 16 49 56

A few additional notes:
- The run name does appear in the legend
- If the corresponding CSV is downloaded the run is there
- This also seems to happen for metrics in the contour plot

### Steps to reproduce the bug

1. Log runs with the provided code
2. Start the UI
3. View the experiment list plots
4. Add a new plot with Metrics.y on one or both axes
5. Notice that the run with Metrics.y==0 is missing

### Code to generate data required to reproduce the bug

```python
import mlflow

for i in range(5):
    with mlflow.start_run():
        mlflow.log_param("x", i)
        mlflow.log_metric("y", 4-i)
```

### Is the console panel in DevTools showing errors relevant to the bug?

_No response_

### Does the network panel in DevTools contain failed requests relevant to the bug?

_No response_

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13666?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13666/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13666
```

</p>
</details>

### Related Issues/PRs

Fix #13659

### What changes are proposed in this pull request?

Before:

(0, 0) is missing:

image

After:

(0, 0) is not missing:

image

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx:1]: /ai let me know if there is a cleaner way to ignore undefined.

File: mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx [hunk 1]
```diff
@@ -87,10 +87,10 @@ export const RunsScatterPlot = React.memo(
         const xAxisData = xAxis.type === 'METRIC' ? metrics : params;
         const yAxisData = yAxis.type === 'METRIC' ? metrics : params;
 
-        const x = xAxisData?.[xAxis.key]?.value || undefined;
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx:90]: 0 is a falsy value in JS.

File: mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx [hunk 2]
```diff
@@ -87,10 +87,10 @@ export const RunsScatterPlot = React.memo(
...
-        const y = yAxisData?.[yAxis.key]?.value || undefined;
+        const x = xAxisData?.[xAxis.key]?.value;
+        const y = yAxisData?.[yAxis.key]?.value;
 
-        if (x && y) {
+        if (x !== undefined && y !== undefined) {
```
Comment (daniellok-db) [inline mlflow/server/js/src/experiment-tracking/components/runs-charts/components/RunsScatterPlot.tsx:93]:
```suggestion
        if (!isNil(x) && !isNil(y)) {
```

to catch `null` and `undefined` (import from lodash)

**Review by daniellok-db — APPROVED** [review]
lgtm after comment is addressed!

**Pull request merged**